node/core/dispute-coordinator/src/lib.rs

```diff
@@ -444,18 +444,27 @@ async fn handle_incoming(
 			let _ = rx.send(collect_active(recent_disputes, now));
 		}
 		DisputeCoordinatorMessage::QueryCandidateVotes(
-			session,
-			candidate_hash,
+			query,
 			rx
 		) => {
-			let candidate_votes = db::v1::load_candidate_votes(
-				store,
-				&config.column_config(),
-				session,
-				&candidate_hash,
-			)?;
-
-			let _ = rx.send(candidate_votes.map(Into::into));
+			let mut query_output = Vec::new();
+			for (session_index, candidate_hash) in query.into_iter() {
+				if let Some(v) = db::v1::load_candidate_votes(
+					store,
+					&config.column_config(),
+					session_index,
+					&candidate_hash,
+				)? {
+					query_output.push(v.into());
+				} else {
+					return Err(
```

Comment:
@rphmeier Here is the modification that ensures we only respond if every `(session_index, candidate_hash)` exists in the db or we return an error.

Reply:
This is returning an io Error from the entire subsystem. We should probably just log something and continue. There's no need to break the message processing loop.

Reply:
It's important to make a distinction between errors the subsystem encounters while processing requests and requests the subsystem can't fulfill, but is fine with being able to skip those. As written, any request referencing missing data could interrupt the subsystem and potentially take it down.